src/SoftDeleteable/Query/TreeWalker/SoftDeleteableWalker.php

```diff
@@ -20,14 +24,50 @@
  */
 class SoftDeleteableWalker extends SqlWalker
 {
+    /**
+     * @var Connection
+     *
+     * @deprecated to be removed in 4.0, use the getConnection method instead
+     */
     protected $conn;
+
+    /**
+     * @var AbstractPlatform
+     *
+     * @deprecated to be removed in 4.0, fetch the platform from the connection instead
+     */
     protected $platform;
```

Comment:
Since the connection is available through a getter, there isn't a strong need to hold a reference to the platform.  It can be fetched as needed at runtime.